clients/src/main/java/org/apache/kafka/common/utils/Time.java

```diff
@@ -32,8 +33,29 @@
     public long nanoseconds();
 
     /**
+     * Returns time in milliseconds as does System.currentTimeMillis(),
+     * but uses elapsed time from an arbitrary epoch more like System.nanoTime().
+     * The difference is that if somebody changes the system clock,
+     * Time.currentElapsedTime will change but nanoTime won't.
+     * @return  The time in milliseconds from some arbitrary point in time.
+     */
+    public long currentElapsedTime();
+
+    /**
+     * Explicitly returns system dependent current wall time.
+     * @return Current time in msec.
+     */
+    public long currentWallTime();
+
+    /**
      * Sleep for the given number of milliseconds
      */
     public void sleep(long ms);
 
+    /**
+     * This is to convert the elapsedTime to a Date.
+     * @return A date object indicated by the elapsedTime.
+     */
+    public Date elapsedTimeToDate(long elapsedTime);
```

Comment:
Not sure whether we should be returning `java.util.Date` as it's a somewhat deprecated class as of Java 8. Maybe we can return a long and the caller can decide to create a `java.util.Date` or anything else?